core/src/database/migrations.cpp

```diff
@@ -955,5 +955,14 @@ namespace ledger {
             sql << "DROP TABLE cosmos_currencies";
 
         }
+
+        template <> void migrate<21>(soci::session& sql, api::DatabaseBackendType type) {
+            sql << "ALTER TABLE bitcoin_accounts ADD balance VARCHAR(255) DEFAULT '0'";
```

Comment:
Hm. That makes the core library even more stateful as possible. I’m strongly against that.

Reply:
What do you mean by more stateful ? 
The current implementation is very stateful, it computes the balance and put it in a memory cache. Therefore with multiple instances on the same DB you end up with different results because the library keeps the state. The change makes the state handled by the database instead of the core library. Now it just read the value in database and this value is only written by the synchronization algorithm.

Reply:
Also, this denormalizes the DB.

Reply:
@pollastri-pierre  How does this change relates to RBF?

Reply:
I'll remove this part, it's another fix which does not concern RBF. 
However I think we'll have to move all integration to this kind of logic:
 - Synchronization is the only process which can write data
 - Getter should be as simple as possible and return result based on state holder (i.e. the DB)
Balances should be moved to the synchronization part because there are heavily hit by clients and require a cache, also there are part of the state of the account which is updated by the synchronization. Only remaining call which can stay lazy are the one related to transaction creation (basically fee computation and sequence/nonce)

Reply:
+1 for moving it to another PR. About heavy operations - I am ok to have a cache. And maybe in this case it is also OK to populate at during write (synchonization) 🤔

Reply:
I'll still need to make a minor change to the balance cache system, because it's failing tests. The cache is not invalidated after a synchro.